test/components/componentTests.ts

```diff
@@ -78,6 +78,17 @@ describe("Component", () => {
       svg.remove();
     });
 
+    it("remove dom elements in previous svg element when switching element", () => {
+      c.anchor(svg);
+      let svg2 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      c.anchor(svg2);
+      assert.isFalse((<Element>svg.node()).hasChildNodes(), "previous svg element should not have any childnodes");
+      assert.isTrue((<Element>svg2.node()).hasChildNodes(), "new svg element should have childnodes.");
```

Comment:
Can you try checking if the `svg` is empty instead to avoid the annoying cast?  If that doesn't work, this is fine it looks like